internal/dag/gatewayapi_processor.go

```diff
@@ -1154,15 +1155,23 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
 			}
 		}
 
+		// Priority is used to ensure if there are multiple matching route rules
+		// within an HTTPRoute, the one that comes first in the list has
+		// precedence. We subtract the index of the rule from MaxUint8 to ensure
+		// rules that come first have a higher priority. All dag.Routes generated
+		// from this HTTPRoute rule matches have the same priority.
+		priority := math.MaxUint8 - uint8(ruleIndex)
```

Comment:
an alternative that would mean we don't have to change a bunch of tests would be to say a lower value means a higher priority

this would then also mean the side-effect of HTTPRoute generated dag.Routes don't automatically get more precedence than conflicting HTTPProxy generated ones

Reply:
(i havent changed the dag package unit tests yet for this reason)

Reply:
Yeah I think this works fine, and a lot fewer tests to have to add a non-zero priority to

Reply:
changed to the inverse, lower number means higher priority